addons/event_crm/models/event_crm.py

```diff
@@ -0,0 +1,143 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models, _
+from odoo.tools.safe_eval import safe_eval
+from odoo.osv import expression
+
+
+class EventCrmRule(models.Model):
+    """
+    Rules to generate a lead (crm.lead) from an event registration (event.registration).
+    When registrations are created, we will check if they check the rules previously created
+    in order to link to them at one or more leads depending on what type of creation the rule
+    has been set.
+    A rule can have 4 types of restrictions:
+        - The company of the event link to the registration
+        - The type of the event
+        - The event it self
+        - A domain set on event.registration
+    When conditions are met, a lead is created with some information prefilled based on the rule
+    and on the registration metting that rule (name, contact_name, email, ...)
+    """
+    _name = "event.crm.rule"
+    _description = "Event CRM Rules"
+
+    # Rule configuration field
+    name = fields.Char('Rule Name', required=True)
+    active = fields.Boolean('Active', default=True)
+    lead_creation_type = fields.Selection([
+        ('one_per_attendee', 'One per Attendee'), ('one_per_order', 'One per Order')],
+        string='Create', default='one_per_attendee', required=True,
+        help='One per Attendee : One lead is created for each attendee created (default):\n'
+            '\te.g. : I go to the "Salon de l\'Auto" page and register for me and 3 friends, 4 attendees are created and 4 leads are created. [Rather B2C]\n'
+            'One per Order : We create a single lead per "registration" (="ticket batch"):\n'
+            '\te.g. : I go to the Odoo Experience page and register for me and 3 colleagues, 4 attendees are created but a single lead is created.  [Rather B2B]')
+
+    """
```

Comment:
Linting: you can move generic comments about the class as docstring class (aka, comment right below Class definition). That way we have a nice list of fields without comments between them.